src/AWSS3.jl

```diff
@@ -538,8 +538,70 @@ function s3_list_objects(aws::AWSConfig, bucket, path_prefix=""; delimiter="/",
     end
 end
 
-s3_list_objects(a...) = s3_list_objects(default_aws_config(), a...)
+s3_list_objects(a...; kwargs...) = s3_list_objects(default_aws_config(), a...; kwargs...)
 
+"""
+    s3_list_objects_v2([::AWSConfig], bucket, [path_prefix]; delimiter="/", max_items=1000)
+
+[List Objects](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGET.html)
+in `bucket` with optional `path_prefix`.
+
+Returns an iterator of `Dict`s with keys `Key`, `LastModified`, `ETag`, `Size`,
+`Owner`, `StorageClass`.
+
+This uses the `ListObjectV2` function call.
+"""
+function s3_list_objects_v2(aws::AWSS3.AWSConfig, bucket, path_prefix=""; delimiter="/", max_items=nothing)
+    return Channel() do chnl
+        more = true
+        num_objects = 0
+        contoken = ""
+
+        while more
+            q = Dict{String, String}("list-type"=>"2")
+            if path_prefix != ""
+                q["prefix"] = path_prefix
+            end
+            if delimiter != ""
+                q["delimiter"] = delimiter
+            end
+            if contoken ≠ ""
+                q["continuation-token"] = contoken
+            end
+            if max_items !== nothing
+                q["max-keys"] = string(max_items - num_objects)
+            end
+
+            @repeat 4 try
+                # Request objects
+                r = s3(aws, "GET", bucket; query = q)
+
+                # Add each object from the response and update our object count / marker
+                if haskey(r, "Contents")
+                    l = isa(r["Contents"], Vector) ? r["Contents"] : [r["Contents"]]
+                    for object in l
```

Comment:
Instead of iterating over these objects, could we not just `put!` the vector into the channel? (Not sure if this is possible).

And then just do `num_objects += size(l)[1]`

Reply:
Wouldn't the xml_dict() function still need to be applied to each element in that vector before `put!`-ing the vector to the channel?